streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -142,6 +145,7 @@ public <K, V> KTable<K, V> table(final String topic,
             .withMaterializedInternal(materialized)
             .withProcessorParameters(processorParameters)
             .build();
+        tableSourceNode.setOutputVersioned(materialized.storeSupplier() instanceof VersionedBytesStoreSupplier);
```

Comment:
Should we make `setOutputVersioned` a `withOutputVersioned` builder method like all the others?